realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -797,6 +829,28 @@ final Builder schema(Class<? extends RealmModel> firstClass, Class<? extends Rea
             return this;
         }
 
+        /**
+         * Sets whether or not users are allowed to perform calls to {@link Realm#executeTransaction} from the UI thread.
+         * <p>
+         * <b>Note: Realm does not allow synchronous transactions to be run on the main thread unless users explicitly opt in
+         * with this method.</b>
+         */
+        public Builder allowWritesOnUiThread(boolean allowWritesOnUiThread) {
+            this.allowWritesOnUiThread = allowWritesOnUiThread;
+            return this;
+        }
+
+        /**
+         * Sets whether or not RealmQueries are allowed from the UI thread.
+         * <p>
```

Comment:
I would also add a line about being cautious about doing this and using `REalmQuery.findAllAsync()` is the preferred approach.